NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops
# Ja deleteacategory ticket11

Description
Trello Ticket: [#11 - Delete A Category](https://trello.com/c/fQmEVNkg/11-11-delete-a-category)
Created code to allow a logged in user to Delete a Category.  
Created the following files and methods:
Added Delete button in view/category/index.cshtml
Added GET/POST Delete methods in Category Controller
Added Delete Category GetCategryById in Category Repository
Added Delete.cshtml file in view/category
Added "NotDelete" to the Post Delete method in Category Controller.
Added a NoDelete.cshtml file toe the View/Category folder that includes the message that "This category cannot be deleted".

Type of change
 New feature (non-breaking change which adds functionality)
  
Testing Instructions
git add and git commit whatever you're working on on your branch
git checkout main
git fetch --a
git checkout ja-deleteacategory-ticket11
If you don't see the most recent version of my files, git pull origin ja-deleteacategory-ticket11
If you haven't created the Tabloid MVC database, please do so.
Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
Click the execute with debugger solid green button to launch the app
In browser, please go to Login on the nav bar and log in with this email: admin@example.com
This should redirect you back to the home page.
Click on Category Management in NavBar
View the list of available Categories (categories should appear in alphabetical order)
(List of Categories should include the list below):

CloseMagic
Cooking
Cthulhu Sightings
Entertainment
Fun
History
Home and Garden
Improv
Movies
Music
Politics
Regrets
Science
Technology

When a user is on the Category list page
They select the `Delete` button (which is in the image of a trash can) beside any category listed
Then they should be prompted to confirm deletion
If the user wants to confirm deletion they hit the delete button 
Then the Category should be removed from the system
And the user should be directed back to the Category list
If the user decides not to confirm deletion - they should click return to list and will be redirected to the Category list 
And the Category should not be removed from the system/list
ADDED INSTRUCTION NOTE AFTER FIX
If user chooses a category to delete that has already been utilized in a post or other feature, then it will give them a message that they cannot delete that category and when the user goes back to the list the category is still there.  
to test this the user should choose the category Technology and try to delete it.  It should trigger the cannot delete message and once the user returns to the list it Technology should still be there.


 **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

# Checklist:
